src/tallies.cpp

```diff
@@ -302,25 +346,89 @@ void Tallies::update_avg_and_var(double x, double& x_avg, double& x_var) {
             ((x_var_old) / (dgen - 1.));
 }
 
+void make_tallies_filter(Tallies& tallies, const YAML::Node& node) {
+  if (!node["tally-filters"]) {
+    fatal_error("tally-filter are not given to evaluate the tallies.");
+  }
+
+  // position-filters
+  if (node["tally-filters"]["position-filters"]) {
+    const YAML::Node& position_filter_nodes =
+        node["tally-filters"]["position-filters"];
+    for (std::size_t i = 0; i < position_filter_nodes.size(); i++) {
+      if (!position_filter_nodes[i]["id"] ||
+          !position_filter_nodes[i]["id"].IsScalar()) {
+        fatal_error(
+            "Invalid \"id\" is given for position filter in tally-filters, the "
+            "possible error entry is " +
+            std::to_string(i) + " from top.");
+      }
+      std::size_t id = position_filter_nodes[i]["id"].as<std::size_t>();
+
+      if (!position_filter_nodes[i]["position-filter-type"] ||
+          !position_filter_nodes[i]["position-filter-type"].IsScalar()) {
+        fatal_error("Invalid entry for the position-filter-type for " +
+                    std::to_string(id) + " id.");
+      }
+      std::string position_filter_name =
+          position_filter_nodes[i]["position-filter-type"].as<std::string>();
+
+      if (position_filter_name == "cylinder-filter") {
+        auto cylinder_filter =
+            make_cylinder_position_filter(position_filter_nodes[i]);
+        tallies.add_cylinder_filter_set(id, cylinder_filter);
+        tallies.add_position_filter_set(id, cylinder_filter);
+      } else {
+        auto cartesian_filter = make_cartesian_filter(position_filter_nodes[i]);
+        tallies.add_cartesian_filter_set(id, cartesian_filter);
+        tallies.add_position_filter_set(id, cartesian_filter);
+      }
+    }
+  }
+
+  // energy-filters
+  if (node["tally-filters"]["energy-filters"]) {
+    const YAML::Node& energy_filter_node =
+        node["tally-filters"]["energy-filters"];
+    for (std::size_t i = 0; i < energy_filter_node.size(); i++) {
+      if (!energy_filter_node[i]["id"] ||
+          energy_filter_node[i]["id"].IsScalar()) {
```

Comment:
```suggestion
          !energy_filter_node[i]["id"].IsScalar()) {
```
It is often safter to say `energy_filter_node[i]["id"].IsScalar() == false`, as it is easy to miss the `!`.